tenprintcover.py

```diff
@@ -498,25 +498,50 @@ def drawShape(c, x, y, s):
         else:
             assert not "Implement."
 
-
+    # If the text is long, use a smaller font size. 
+    def scale_font(text, font_name, font_properties):
+        (font_size, font_slant, font_weight) = font_properties
+        width = len(text) * font_size
+        if width > cover_width * 7:   # This is an empirical, unintelligent, heuristic.
+            return  (font_size * 0.8, font_slant, font_weight)
+        else:
+            return font_properties
+    
+    # Return a font appropriate for the text. Uses Noto CJK if text contains letters of
+    # Simplified Chinese, Traditional Chinese, Japanese, and Korean (CJK), otherwise Noto Sans. 
+    # http://www.unicode.org/faq/han_cjk.html
+    def select_font(text):
+        for char in text:
+            if ord(char) >= 0x4E00:
+                return 'Noto Sans CJK SC'
+        return 'Noto Sans'
```

Comment:
I think the [FontTools solution](http://stackoverflow.com/questions/4458696/finding-out-what-characters-a-font-supports#19438403) is the best way to address font selection, but I have too many other fish to fry at the moment to work on it.